src/guide/agent-functions/interacting-with-environment.rst

```diff
@@ -93,9 +118,32 @@ Example usage is shown below:
         // Other behaviour code
         ...
     }
+
+  .. code-tab:: py Agent Python
+
+    @pyflamegpu.agent_function
+    def ExampleFn(message_in: pyflamegpu.MessageNone, message_out: pyflamegpu.MessageNone):
+        # Get the root of the 3x3x3 environment macro property 'location' and store it in a variable of the same name
+        location = pyflamegpu.environment.getMacroPropertyUInt("location", 3, 3, 3)
+        # Notify our location, of our presence and store how many other agents were there before us in `location_count`
+        location_count = location[0][1][2]+=1
+        
+        
+        # Get the root of the float environment macro property 'swap' and store it in a variable of the same name
+        swap = pyflamegpu.environment.getMacroPropertyFloat("swap")
+        # Fetch and replace the value present in swap
+        location_count = swap.exchange(12.0f)
+        
+        # Directly accessing the value of either macro property now in the same agent function would cause a race condition
+        # location_val = location[0][0][0] # DeviceError!
+        # swap_val = swap # DeviceError!
+
+        # Other behaviour code
+        ...
+    }
     
 .. warning::
-  Be careful when using :class:`DeviceMacroProperty<flamegpu::DeviceMacroProperty>`. When you retrieve an element e.g. ``location[0][0][0]`` (from the example above), it is of type :class:`DeviceMacroProperty<flamegpu::DeviceMacroProperty>` not ``unsigned int``. Therefore you cannot pass it directly to functions which take generic arguments such as ``printf()``, as it will be interpreted incorrectly. You must either store it in a variable of the correct type which you instead pass, or explicitly cast it to the correct type when passing it e.g. ``(unsigned int)location[0][0][0]`` or ``static_cast<unsigned int>(location[0][0][0])``.
+  Be careful when using :class:`DeviceMacroProperty<flamegpu::DeviceMacroProperty>`. When you retrieve an element e.g. ``location[0][0][0]`` (from the example above), it is of type :class:`DeviceMacroProperty<flamegpu::DeviceMacroProperty>` not ``unsigned int``. Therefore you cannot pass it directly to functions which take generic arguments such as ``printf()``, as it will be interpreted incorrectly. You must either store it in a variable of the correct type which you instead pass, or explicitly cast it to the correct type when passing it e.g. ``(unsigned int)location[0][0][0]`` or ``static_cast<unsigned int>(location[0][0][0])`` (or ``int(location[0][0][0])`` in Python).
```

Comment:
is `int` correct given the example is for an unsigned integer, so `numpy.uint`?

Reply:
Ahh yes. `int` is a catchall for all integer types in Python and the `int()` function is built in. There is no `uint()` equivalent.

Reply:
hence the suggestion to use `numpy.uint` as previously refferred to in other sections of the docs. 

Surely casting to `int` will impliclty convert the value into a signed integer of sufficient width? 

If you don't want to veryify this behaves correctly by specifying an integer `> INT_MAX` but `< UINT_MAX` then changing the example here to instead use an `int` instead of an `unsigned int` would be the easy but less helpful fix.

Reply:
ok have moved it to a a `numpy.uint` (I don't think its strictly necessary but doesn't hurt)